### DISCUSSION TITLE
STL with HTTPS/TLS

### DISCUSSION THREAD

**Discussion by jkalaise** [cisco-system-traffic-generator/trex-core#1093] [Q&A]
Team,

I am able to generate TCP/HTTP packet on stateless using below way,

base_pkt = Ether(dst="18:5a:58:4c:75:62")/Dot1Q(vlan = 2509)/IP(src="97.0.0.1",dst="94.0.0.1")/TCP()/"GET /3384 HTTP/1.1\r\nHost: [www.google.com\r\nConnection](http://www.google.com%5Cr%5CnConnection): Keep-Alive\r\nUser-Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1; SV1; .NET CLR 1.1.4322; .NET CLR 2.0.50727)\r\nAccept: /\r\nAccept-Language: en-us\r\nAccept-Encoding: gzip, deflate, compress\r\n\r\n"

    return STLStream(packet = STLPktBuilder(pkt = base_pkt/pad,vm = vm),
                     mode = STLTXCont( pps=1 ),
                     isg=0.1)

Now when I am trying to do the similar one for HTTPS, I am not able to construct with STLPktBuilder.

p = TLSRecord() / TLSHandshake() / TLSClientHello(cipher_suites=[TLSCipherSuite.RSA_WITH_AES_128_CBC_SHA])
stream = STLPktBuilder(pkt = p)
File "stl/610_2_https.py", line 43, in get_streams
return [ self.create_stream() ]
File "stl/610_2_https.py", line 17, in create_stream
base_pkt = Ether(dst="8C:47:BE:8C:56:95")/Dot1Q(vlan = 2506)/IP(src="94.0.0.1",dst="97.0.0.1")/TCP()/TLSClientHello(cipher_suites=[TLSCipherSuite.RSA_WITH_AES_128_CBC_SHA])
NameError: name 'TLSClientHello' is not defined

Could you please me on how to achieve this?

Regards,
Jasmine.K

**Comment by hhaim** [comment]
@jkalaise this is a scapy issue. I would ask it in scapy forum. from TRex prospective the builder is just creating a byte stream that will be sent using STL. I think that for the Hello packet it is ok but for the full TLS protocol you won't be able to use scapy

**Comment by jkalaise** [comment]
Thank you for getting back Hanoh. Kindly update back incase there is any
resolution from scapy team.

Regards,
Jasmine.K
Those who wait on the Lord shall renew their strength; they shall mount up
with wings like eagles. —Isaiah 40:31
